regression/pages/common/api_clients.py

```diff
@@ -0,0 +1,260 @@
+
+
+class LmsApiClient(object):
+    """
+    API class for accessing different APIs
+    """
+
+    def __init__(self, host=None):
+        """
+        Instantiate the api class.
+        """
+        self.host = host or URL_WITHOUT_AUTH
+        self.session = requests.Session()
+        self.session.auth = (AUTH_USERNAME, AUTH_PASSWORD)
+        self.login_response = None
+
+    def _post_headers(self, x_csrf=None):
+        """
+        Default HTTP headers dict.
+        Args:
+            x_csrf:
+        """
+        return {
+            'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+            'Accept': '*/*',
+            'X-Requested-With': 'XMLHttpRequest',
+            'Referer': self.host,
+            'X-CSRFToken': x_csrf,
+        }
+
+    def create_base_session(self):
+        """
+        Open target site and create a base session
+        """
+        response = self.session.get(self.host)
+        check_response(response)
+        self.session.cookies = response.cookies
+        self.session.headers = self._post_headers(
+            response.cookies['csrftoken']
+        )
+
+    def create_login_session(self, user_email, user_password):
+        """
+        Login to site using base session and replace base session with the
+        logged in session
+        Args:
+            user_email:
+            user_password:
+        """
+        self.create_base_session()
+        login_url = self.host + 'user_api/v1/account/login_session/'
+        payload = {
+            'email': user_email,
+            'password': user_password,
+            'remember': 'false'
+        }
+        response = self.session.post(login_url, data=payload)
+        check_response(response)
+        self.session.cookies = response.cookies
+        self.session.headers = self._post_headers(
+            response.cookies['csrftoken']
+        )
+        self.login_response = response
+
+    @property
+    def login_response_cookies(self):
+        """
+        get cookie info from response headers
+        Returns:
+
+        """
+        set_cookie = self.login_response.headers['Set-Cookie']
+        if not set_cookie:
+            raise ApiException('Login response cookie not found')
+        return Cookie.SimpleCookie(set_cookie)
+
+    @property
+    def user_name(self):
+        """
+        Extract user name from response
+        Returns:
+            username
+        """
+        user_info = self.login_response_cookies['stage-edx-user-info'].value
+        user_info_dict = json.loads(user_info)
+        if not user_info_dict:
+            raise ApiException('User Info Dictionary not found')
+        return user_info_dict['username']
+
+    def change_enrollment(self, course_id, enrollment_action):
+        """
+        Change enrollment
+        Args:
+            course_id:
+            enrollment_action:
+        """
+        change_enrollment_url = self.host + 'change_enrollment'
```

Comment:
They end with a slash *now* but someone may make a change later in which they don't. Or will set an environment variable where they don't. So you just want to think through that and assume someone makes a manual mistake somewhere. (If it were me, I would not know if I need a slash or not, so it's a bit arbitrary what I might set locally.)

You need either instructions to end with a slash, or (better) a utility that ensures `self.host` ends with a slash.